drivers/entropy/entropy_ll_nrfxlib.c

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <entropy.h>
+#include <soc.h>
+#include <device.h>
+#include <irq.h>
+#include <kernel.h>
+#include <kernel_includes.h>
+#include <ble_controller.h>
+#include <ble_controller_soc.h>
+
+#include <multithreading_lock.h>
+
+#define DEV_DATA(dev) \
+	((struct rng_driver_data *)(dev)->driver_data)
+
+/* The RNG driver data. */
+struct rng_driver_data {
+	struct k_sem sem_sync; /* Used to fill the pending client buffer with new
+	                        * random values after RNG IRQ is triggered. */
+};
+
+static struct rng_driver_data rng_data;
+
+static int rng_driver_get_entropy(struct device *dev, u8_t *buf, u16_t len)
+{
+	__ASSERT_NO_MSG(&rng_data == DEV_DATA(dev));
```

Comment:
What is the sense of putting this pointer into a dev structure if we are even using rng_data directly in the code below?

Reply:
I suppose the code below should use the `dev*` instead and the assertion should remain to ensure that these APIs are not called before the device has been initialized.